### PR TITLE
Properly parse network addresses.

### LINKED ISSUE #1510: deno.dial fails
**command:**
`RUST_BACKTRACE=1 deno httpClient.ts --reload --allow-net`

**script:**

```
import * as deno from "deno";

(async () => {
  try {
    const res = await deno.dial('tcp',':80');
  } catch (e) {
    console.log(e);
  }
})();
```
also tried 'deno.land:http', deno.land:80' as address, got the same output.

**output:**
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: AddrParseError(())', libcore/result.rs:1009:5
stack backtrace:
   0: <std::sys::unix::backtrace::tracing::imp::UnwindError as core::fmt::Display>::fmt
   1: <std::env::ArgsOs as core::iter::iterator::Iterator>::size_hint
   2: std::panicking::take_hook
   3: std::panicking::take_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::begin_panic_fmt
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: unicode_normalization::normalize::is_hangul_syllable
   9: unicode_normalization::normalize::is_hangul_syllable
  10: unicode_normalization::normalize::is_hangul_syllable
  11: unicode_normalization::normalize::is_hangul_syllable
  12: <core::sync::atomic::AtomicIsize as core::fmt::Debug>::fmt
  13: <core::sync::atomic::AtomicIsize as core::fmt::Debug>::fmt
  14: <core::sync::atomic::AtomicIsize as core::fmt::Debug>::fmt
  15: <core::sync::atomic::AtomicIsize as core::fmt::Debug>::fmt
fatal runtime error: failed to initiate panic, error 5
Abort trap: 6
```

### PR DESCRIPTION
**Pull request by ry**
Fixes #1510

### CODE REVIEW

File: src/resolve_addr.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+use std::net::ToSocketAddrs;
+
+/// Go-style network address parsing. Returns a future.
+/// Examples:
+/// "192.0.2.1:25"
+/// ":80"
+/// "[2001:db8::1]:80"
+/// "198.51.100.1:80"
+/// "deno.land:443"
+pub fn resolve_addr(address: &str) -> ResolveAddrFuture {
```
Comment (piscisaureus) [inline src/resolve_addr.rs]: How do you decide whether to resolve ipv4/ipv6 or either ?

File: src/resolve_addr.rs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      }
+    }
+  }
+}
+
+fn split<'a>(address: &'a str) -> Option<(&'a str, u16)> {
+  address.rfind(":").and_then(|i| {
+    let (a, p) = address.split_at(i);
+    // Default to localhost if given just the port. Example: ":80"
+    let addr = if a.len() > 0 { a } else { "127.0.0.1" };
```
Comment (piscisaureus) [inline src/resolve_addr.rs]: Does go also default to IPv4 and localhost? Other valid defaults might be `::1` and `0.0.0.0`.

Comment (ry) [reply src/resolve_addr.rs]: https://github.com/golang/go/blob/7cbfa55b5d17c8deaecff05e4221f828467cfa97/src/net/ipsock.go#L275-L277

It appears to default to "0.0.0.0". I've updated the code. Good catch.

File: src/resolve_addr.rs (hunk 1 above)
Comment (ry) [reply src/resolve_addr.rs]: Rust's resolution scheme to_socket_addrs() decides that. This procedure I've made is a bit hacky and just uses the first ip returned.

**Pull request merged**